### PR TITLE
Add operator crds

### PR DESCRIPTION
**Pull request by soltysh**
/hold

### CODE REVIEW

**Comment by soltysh** [comment]
/hold cancel
/assign @deads2k @damemi 
I've added most (but not all) operator CRDs and fixed them following the patterns we've set last week. I'll send out an email to aos-devel shortly. ptal in the mean time.

**Comment by deads2k** [comment]
looks plausible to me.

@damemi owns lgtm
/assign @damemi

**Comment by damemi** [comment]
I think some of these operators have `yaml-merge-patch` files like https://github.com/openshift/cluster-kube-apiserver-operator/blob/master/manifests/0000_20_kube-apiserver-operator_01_config.crd.yaml-merge-patch, so in those cases we also need them to move with the CRDs to ensure the patch is applied when running update-codegen

**Comment by soltysh** [comment]
@damemi of all the operators I've added in this PR only kas-o, kcm-o ks-o and ocm-o have CRD patches and they are ensuring there's a pattern check on top of `.spec.managementState` which is currently by default applied consistently from [here](https://github.com/openshift/api/blob/master/operator/v1/types.go#L27).

File: operator/v1/0000_20_kube-apiserver-operator_01_config.crd.yaml [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+              description: logLevel is an intent based logging for an overall component.  It
+                does not give fine grained control, but it is a simple way to manage
+                coarse grained logging choices that operators have to interpret for
+                their operands.
+              type: string
+            managementState:
+              description: managementState indicates whether and how the operator
+                should manage the component
+              type: string
+              pattern: ^(Managed|Unmanaged|Force|Removed)$
```
Comment (sttts) [inline operator/v1/0000_20_kube-apiserver-operator_01_config.crd.yaml]: this is wrong

Comment (soltysh) [reply operator/v1/0000_20_kube-apiserver-operator_01_config.crd.yaml]: Last commit adds patches where they should exist, iow. here. The question remains whether kcm-o and ks-o should have a validation similar to kas-o, iow. only `Managed|Force` allowed or like the default generated. @sttts did some tests and nothing happened when setting kcm-o state to `Removed`, @deads2k thoughts?

Comment (sttts) [reply operator/v1/0000_20_kube-apiserver-operator_01_config.crd.yaml]: kcm and ks should be able to recover when set to Removed, and then to Managed by the user. So we should survive that. With ka this is different why we added the extra validaiton.

File: operator/v1/0000_20_kube-apiserver-operator_01_config.crd.yaml-merge-patch [hunk 2]
```diff
@@ -0,0 +1,8 @@
+spec:
+  validation:
+    openAPIV3Schema:
+      properties:
+        spec:
+          properties:
+            managementState:
+              pattern: "^(Managed|Force)$"
```
Comment (sttts) [inline operator/v1/0000_20_kube-apiserver-operator_01_config.crd.yaml-merge-patch]: you have to regenerate for this to be applied.

File: operator/v1/0000_20_kube-apiserver-operator_01_config.crd.yaml (hunk 1 above)
Comment (deads2k) [reply operator/v1/0000_20_kube-apiserver-operator_01_config.crd.yaml]: Without a kcm and without a ks, how would the operator pods (from deployments) that would re-install these static pods be created and scheduled?  Removed needs to be ignored to ensure that the system remains running.

**Comment by damemi** [comment]
/lgtm
Looks like we only needed the patch for kas, the others just set what is now the default

**Comment by soltysh** [comment]
> Looks like we only needed the patch for kas, the others just set what is now the default

Nope, discussed that with @deads2k we want the same limited set for all 3 of them.

**Pull request merged**